Sources/MessagingInApp/Gist/Managers/MessageQueueManager.swift

```diff
@@ -1,152 +1,113 @@
 import CioInternalCommon
 import Foundation
+import ReSwift
 import UIKit
 
+// sourcery: InjectRegisterShared = "MessageQueueManager"
+// sourcery: InjectSingleton
 class MessageQueueManager {
     private let logger: Logger
-    var interval: Double = 600
+    private let inAppMessageManager: InAppMessageManager
+    private let queueManager: QueueManager
+    private let threadUtil: ThreadUtil
+
+    private var inAppMessageStoreSubscriber: InAppMessageStoreSubscriber?
     private var queueTimer: Timer?
-    // The local message store is used to keep messages that can't be displayed because the route rule doesnt match.
-    var localMessageStore: [String: Message] = [:]
 
-    init() {
-        self.logger = DIGraphShared.shared.logger
+    init(logger: Logger, inAppMessageManager: InAppMessageManager, queueManager: QueueManager, threadUtil: ThreadUtil) {
+        self.logger = logger
+        self.inAppMessageManager = inAppMessageManager
+        self.queueManager = queueManager
+        self.threadUtil = threadUtil
+
+        subscribeToInAppMessageState()
     }
 
-    func setup(skipQueueCheck: Bool = false) {
+    private func subscribeToInAppMessageState() {
+        inAppMessageStoreSubscriber = InAppMessageStoreSubscriber { [weak self] state in
+            guard let self else { return }
+
+            let newPollingInterval = state.pollInterval
+            self.setupPollingAndFetch(skipMessageFetch: true, pollingInterval: newPollingInterval)
+        }
+        if let subscriber = inAppMessageStoreSubscriber {
+            inAppMessageManager.subscribe(keyPath: \.pollInterval, subscriber: subscriber)
+        }
+    }
+
+    func setupPollingAndFetch(skipMessageFetch: Bool, pollingInterval: Double) {
         queueTimer?.invalidate()
         queueTimer = nil
 
-        queueTimer = Timer.scheduledTimer(
-            timeInterval: interval,
-            target: self,
-            selector: #selector(fetchUserMessages),
-            userInfo: nil,
-            repeats: true
-        )
+        threadUtil.runMain {
+            self.queueTimer = Timer.scheduledTimer(
+                timeInterval: pollingInterval,
+                target: self,
+                selector: #selector(self.fetchUserMessages),
+                userInfo: nil,
+                repeats: true
+            )
+        }
 
-        if !skipQueueCheck {
+        if !skipMessageFetch {
             // Since on app launch there's a short period where the applicationState is still set to "background"
             // We wait 1 second for the app to become active before checking for messages.
-            DispatchQueue.main.asyncAfter(deadline: .now() + 1) {
+            threadUtil.runMainAfterDelay(deadline: .now() + 1) {
```

Comment:
now that we are waiting for initializing task to finish, we might not need to delay?